pkg/run/get_kube_client_test.go

```diff
@@ -0,0 +1,57 @@
+package run_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/pflag"
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	runCmd "github.com/weaveworks/weave-gitops/cmd/gitops/beta/run"
+	"github.com/weaveworks/weave-gitops/pkg/logger/loggerfakes"
+	"github.com/weaveworks/weave-gitops/pkg/run"
+	"github.com/weaveworks/weave-gitops/pkg/version"
+)
+
+var _ = Describe("GetKubeClient", func() {
+	var fakeLogger *loggerfakes.FakeLogger
+
+	BeforeEach(func() {
+		fakeLogger = &loggerfakes.FakeLogger{}
+	})
+
+	It("returns kube client", func() {
```

Comment:
This test makes me feel funny, because it's struggling with its scope - it's the only bit of code we have that uses `pflag` (directly, at least) which raises one of my eyebrows, and it specifies the dashboard port which is unrelated to kubeclients.

Does it still pass without the `kubeConfigArgs.AddFlags`? If there's a specific flag you need, can we add just those flags instead of going through all RunCommandFlags?

This suggestion removes what I think is a little bit of a hidden side-effect of this test - it also tests that "our command line arguments don't clash horribly with those understood by kubeconfig", which is valuable but I think that test ought to live in the main command. Perhaps we need a `gitops beta run --dry-run` that sets up kubeclients but doesn't do anything, and we could write a test in `cmd/beta/run` that uses that functionality.

Reply:
Yes, I just copied all flags from the gitops run command to mock them with pflag in order not to call cmd.Flags() to get them.

For the current test I can remove the flags completely. It should pass.

OK, will remove the flags. I thought of adding a direct test for the run command later just have not looked yet at how to do it.